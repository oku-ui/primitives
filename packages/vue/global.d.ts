/// <reference types="vite/client" />
import 'vitest'
import type { AxeMatchers } from 'vitest-axe/matchers'

declare module 'vitest' {
  export interface AsymmetricMatchersContaining extends AxeMatchers {}
  export interface Assertion extends AxeMatchers {}
}
