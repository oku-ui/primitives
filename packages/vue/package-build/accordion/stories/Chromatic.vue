<script setup lang="ts">
import { ref } from 'vue'
import { OkuAccordion, OkuAccordionContent, OkuAccordionHeader, OkuAccordionItem, OkuAccordionTrigger } from '@oku-ui/accordion'

const values = ref(['One', 'Two', 'Three', 'Four'])
</script>

<template>
  <div>
    <h1>Uncontrolled</h1>
    <h2>Single closed</h2>
    <OkuAccordion type="single" class="accordion-root">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Single open</h2>
    <OkuAccordion type="single" class="accordion-root" default-value="Two">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Multiple closed</h2>
    <OkuAccordion type="multiple" class="accordion-root">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Multiple open</h2>
    <OkuAccordion type="multiple" class="accordion-root" :default-value="['One', 'Two']">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>Controlled</h1>
    <h2>Single open</h2>
    <OkuAccordion type="single" class="accordion-root" value="Three">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Multiple open</h2>
    <OkuAccordion type="multiple" class="accordion-root" :value="['Two', 'Three']">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>Disabled (whole)</h1>
    <OkuAccordion type="single" class="accordion-root" disabled>
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>Disabled (item)</h1>
    <h2>Just item</h2>
    <OkuAccordion type="single" class="accordion-root">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item" :disabled="item === 'Two'">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>with `disabled=false` on top-level</h2>
    <OkuAccordion type="single" class="accordion-root" :disabled="false">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item" :disabled="item === 'Two'">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>Force mounted contents</h1>
    <OkuAccordion type="single" class="accordion-root">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content" force-mount>
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>State attributes</h1>
    <h2>Accordion disabled</h2>
    <OkuAccordion type="single" class="accordion-root-attr" default-value="Two" disabled>
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item-attr" :value="item">
        <OkuAccordionHeader class="accordion-header-attr">
          <OkuAccordionTrigger class="accordion-trigger-attr">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content-attr">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Accordion enabled with item override</h2>
    <OkuAccordion type="single" class="accordion-root-attr" default-value="Two" :disabled="false">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item-attr" :value="item" :disabled="['Two', 'Four'].includes(item)">
        <OkuAccordionHeader class="accordion-header-attr">
          <OkuAccordionTrigger class="accordion-trigger-attr">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content-attr">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Accordion disabled with item override</h2>
    <OkuAccordion type="single" class="accordion-root-attr" default-value="Two" :disabled="true">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item-attr" :value="item" :disabled="['Two', 'Four'].includes(item) ? false : undefined">
        <OkuAccordionHeader class="accordion-header-attr">
          <OkuAccordionTrigger class="accordion-trigger-attr">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content-attr">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
  </div>
</template>
