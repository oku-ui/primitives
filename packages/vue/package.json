{
  "name": "vue-packages",
  "type": "module",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "dev": "vite",
    "build": "vite build",
    "preview": "vite preview"
  },
  "dependencies": {
    "@rollup/plugin-alias": "^5.1.0"
  },
  "devDependencies": {
    "@vitejs/plugin-vue": "^5.0.3",
    "@vueuse/core": "^10.7.2",
    "filter-anything": "^3.0.7",
    "typescript": "^5.3.3",
    "unplugin-vue-components": "^0.26.0",
    "vite-plugin-pages": "^0.32.0",
    "vue-router": "^4.2.5",
    "vue-tsc": "^2.0.0"
  }
}
