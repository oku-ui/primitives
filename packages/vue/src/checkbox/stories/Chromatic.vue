<script setup lang="ts">
import { OkuCheckbox, OkuCheckboxIndicator } from '@oku-ui/checkbox'
</script>

<template>
  <h1>Uncontrolled</h1>
  <h2>Unchecked</h2>
  <OkuCheckbox class="checkbox">
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h2>Checked</h2>
  <OkuCheckbox class="checkbox" default-checked>
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h1>Controlled</h1>
  <h2>Unchecked</h2>
  <OkuCheckbox class="checkbox" :checked="false">
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h2>Checked</h2>
  <OkuCheckbox class="checkbox" checked>
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h1>Indeterminate</h1>
  <OkuCheckbox class="checkbox" checked="indeterminate">
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h1>Disabled</h1>
  <OkuCheckbox class="checkbox" default-checked disabled>
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h1>Force mounted indicator</h1>
  <OkuCheckbox class="checkbox">
    <OkuCheckboxIndicator class="checkbox-indicator" force-mount :style="{ height: '20px' }" />
  </OkuCheckbox>

  <h1>State attributes</h1>
  <h2>Unchecked</h2>
  <OkuCheckbox class="checkbox-attr-styles">
    <OkuCheckboxIndicator class="checkbox-attr-styles" />
  </OkuCheckbox>

  <h2>Checked</h2>
  <OkuCheckbox class="checkbox-attr-styles" default-checked>
    <OkuCheckboxIndicator class="checkbox-attr-styles" />
  </OkuCheckbox>

  <h2>Indeterminate</h2>
  <OkuCheckbox class="checkbox-attr-styles" checked="indeterminate">
    <OkuCheckboxIndicator class="checkbox-attr-styles" />
  </OkuCheckbox>

  <h2>Disabled</h2>
  <OkuCheckbox class="checkbox-attr-styles" default-checked disabled>
    <OkuCheckboxIndicator class="checkbox-attr-styles" />
  </OkuCheckbox>

  <h2>Force mounted indicator</h2>
  <OkuCheckbox class="checkbox-attr-styles">
    <OkuCheckboxIndicator class="checkbox-attr-styles" force-mount :style="{ height: '20px' }" />
  </OkuCheckbox>
</template>
