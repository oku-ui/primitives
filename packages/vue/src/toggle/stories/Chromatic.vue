<script setup lang="ts">
import { OkuToggle } from '@oku-ui/toggle'
</script>

<template>
  <h1>Uncontrolled</h1>
  <h2>Off</h2>
  <OkuToggle class="toggle">
    Toggle
  </OkuToggle>

  <h2>On</h2>
  <OkuToggle class="toggle" default-pressed>
    Toggle
  </OkuToggle>

  <h1>Controlled</h1>
  <h2>Off</h2>
  <OkuToggle class="toggle" :pressed="false">
    Toggle
  </OkuToggle>

  <h2>On</h2>
  <OkuToggle class="toggle" pressed>
    Toggle
  </OkuToggle>

  <h1>Disabled</h1>
  <OkuToggle class="toggle" disabled>
    Toggle
  </OkuToggle>

  <h1>State attributes</h1>
  <OkuToggle class="toggle-attr-styles">
    Toggle
  </OkuToggle>
  <OkuToggle class="toggle-attr-styles" disabled>
    Toggle
  </OkuToggle>
</template>
