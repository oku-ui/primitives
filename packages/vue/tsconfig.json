{
  "compilerOptions": {
    "target": "ESNext",
    "jsx": "preserve",
    "jsxImportSource": "vue",
    "lib": [
      "ESNext",
      "DOM",
      "DOM.Iterable"
    ],
    "useDefineForClassFields": true,
    "baseUrl": ".",
    "module": "ESNext",
    "moduleResolution": "bundler",
    "paths": {
      "@/*": [
        "src/*"
      ],
      "@oku-ui/accordion": [
        "src/accordion"
      ],
      "@oku-ui/alert-dialog": [
        "src/alert-dialog"
      ],
      "@oku-ui/arrow": [
        "src/arrow"
      ],
      "@oku-ui/aspect-ratio": [
        "src/aspect-ratio"
      ],
      "@oku-ui/avatar": [
        "src/avatar"
      ],
      "@oku-ui/collection": [
        "src/collection"
      ],
      "@oku-ui/collapsible": [
        "src/collapsible"
      ],
      "@oku-ui/checkbox": [
        "src/checkbox"
      ],
      "@oku-ui/dialog": [
        "src/dialog"
      ],
      "@oku-ui/direction": [
        "src/direction"
      ],
      "@oku-ui/dismissable-layer": [
        "src/dismissable-layer"
      ],
      "@oku-ui/focus-guards": [
        "src/focus-guards"
      ],
      "@oku-ui/focus-scope": [
        "src/focus-scope"
      ],
      "@oku-ui/hover-card": [
        "src/hover-card"
      ],
      "@oku-ui/label": [
        "src/label"
      ],
      "@oku-ui/menu": [
        "src/menu"
      ],
      "@oku-ui/popover": [
        "src/popover"
      ],
      "@oku-ui/popper": [
        "src/popper"
      ],
      "@oku-ui/portal": [
        "src/portal"
      ],
      "@oku-ui/primitive": [
        "src/primitive"
      ],
      "@oku-ui/progress": [
        "src/progress"
      ],
      "@oku-ui/provide": [
        "src/provide"
      ],
      "@oku-ui/radio-group": [
        "src/radio-group"
      ],
      "@oku-ui/roving-focus": [
        "src/roving-focus"
      ],
      "@oku-ui/scroll-area": [
        "src/scroll-area"
      ],
      "@oku-ui/separator": [
        "src/separator"
      ],
      "@oku-ui/slider": [
        "src/slider"
      ],
      "@oku-ui/slot": [
        "src/slot"
      ],
      "@oku-ui/switch": [
        "src/switch"
      ],
      "@oku-ui/tabs": [
        "src/tabs"
      ],
      "@oku-ui/toggle": [
        "src/toggle"
      ],
      "@oku-ui/toggle-group": [
        "src/toggle-group"
      ],
      "@oku-ui/toast": [
        "src/toast"
      ],
      "@oku-ui/toolbar": [
        "src/toolbar"
      ],
      "@oku-ui/tooltip": [
        "src/tooltip"
      ],
      "@oku-ui/use-composable": [
        "src/use-composable"
      ],
      "@oku-ui/utils": [
        "src/utils"
      ],
      "@oku-ui/visually-hidden": [
        "src/visually-hidden"
      ],
      "@oku-ui/presence": [
        "src/presence"
      ]
    },
    "resolveJsonModule": true,
    "allowImportingTsExtensions": true,
    "strict": true,
    "noFallthroughCasesInSwitch": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noEmit": true,
    "isolatedModules": true,
    "skipLibCheck": true
  },
  "references": [
    {
      "path": "./tsconfig.node.json"
    }
  ],
  "include": [
    "src/**/*.ts",
    "src/**/*.tsx",
    "src/**/*.vue",
    "*.d.ts"
  ]
}
