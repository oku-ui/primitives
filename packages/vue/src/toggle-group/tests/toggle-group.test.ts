import { defineComponent } from 'vue'
import { enableAutoUnmount, mount, shallowMount } from '@vue/test-utils'
import type { DOMWrapper, VueWrapper } from '@vue/test-utils'
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
import { axe } from 'vitest-axe'

import { OkuToggleGroup, OkuToggleGroupItem } from '../'

import Single from '../stories/Single.vue'
import Vertical from '../stories/Vertical.vue'
import Multiple from '../stories/Multiple.vue'
import Chromatic from '../stories/Chromatic.vue'

enableAutoUnmount(afterEach)

const onValueChange = vi.fn()

const ToggleGroupTest = defineComponent({
  components: {
    OkuToggleGroup,
    OkuToggleGroupItem,
  },
  props: {
    type: String,
  },
  setup() {
    return {
      onValueChange,
    }
  },
  template: `
    <OkuToggleGroup :type="type" @value-change="onValueChange">
      <OkuToggleGroupItem value="One">One</OkuToggleGroupItem>
      <OkuToggleGroupItem value="Two">Two</OkuToggleGroupItem>
      <OkuToggleGroupItem value="Three">Three</OkuToggleGroupItem>
    </OkuToggleGroup>
  `,
})

describe('okuToggleGroup', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = shallowMount(ToggleGroupTest, {
      attachTo: document.body,
    })
  })

  it('should render OkuToggleGroup correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()

    expect(shallowMount(OkuToggleGroup).html()).toMatchSnapshot()
  })

  it('should render OkuToggleGroupItem correctly', () => {
    const spy = vi.spyOn(globalThis.console, 'warn').mockImplementation(() => { })
    const wrapper = () => shallowMount(OkuToggleGroupItem)

    expect(() => wrapper()).toThrowErrorMatchingSnapshot()

    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuToggleGroup)" not found.')
  })

  /**
   * @vitest-environment jsdom
   */
  it('should have no accessibility violations', async () => {
    expect(await axe(wrapper.element)).toHaveNoViolations()
  })

  it('should log a warning when the type prop is not provided', () => {
    const spy = vi.spyOn(console, 'warn').mockImplementation(() => { })

    wrapper = shallowMount(OkuToggleGroup)

    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: Missing required prop: "type"')
  })

  describe('given a single ToggleGroup', () => {
    let wrapper: VueWrapper
    let one: DOMWrapper<Element>
    let two: DOMWrapper<Element>

    beforeEach(() => {
      wrapper = mount(ToggleGroupTest, {
        props: {
          type: 'single',
        },
        attachTo: document.body,
      })

      one = wrapper.findAll('button').filter(button => button.text() === 'One').at(0)!
      two = wrapper.findAll('button').filter(button => button.text() === 'Two').at(0)!
    })

    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    describe('when clicking `One`', () => {
      beforeEach(() => {
        one.trigger('click')
      })

      it('should have no accessibility violations', async () => {
        expect(await axe(wrapper.element)).toHaveNoViolations()
      })

      it('should change value to `One`', () => {
        expect(onValueChange).toHaveBeenCalledWith('One')
      })

      describe('then clicking `Two`', () => {
        beforeEach(() => {
          two.trigger('click')
        })

        it('should change value to `Two`', () => {
          expect(onValueChange).toHaveBeenCalledWith('Two')
        })

        describe('and clicking `Two` again`', () => {
          beforeEach(() => {
            two.trigger('click')
          })

          it('should change value to empty string', () => {
            expect(onValueChange).toHaveBeenCalledWith('')
          })
        })
      })
    })
  })

  describe('given a multiple ToggleGroup', () => {
    let wrapper: VueWrapper
    let one: DOMWrapper<Element>
    let two: DOMWrapper<Element>

    beforeEach(() => {
      wrapper = mount(ToggleGroupTest, {
        props: {
          type: 'multiple',
        },
        attachTo: document.body,
      })

      one = wrapper.findAll('button').filter(button => button.text() === 'One').at(0)!
      two = wrapper.findAll('button').filter(button => button.text() === 'Two').at(0)!
    })

    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    describe('when clicking `One`', () => {
      beforeEach(() => {
        one.trigger('click')
      })

      it('should have no accessibility violations', async () => {
        expect(await axe(wrapper.element)).toHaveNoViolations()
      })

      it('should change value to `One`', () => {
        expect(onValueChange).toHaveBeenCalledWith(['One'])
      })

      describe('and clicking `One` again`', () => {
        beforeEach(() => {
          one.trigger('click')
        })

        it('should change value to empty array', () => {
          expect(onValueChange).toHaveBeenCalledWith([])
        })
      })

      describe('then clicking `Two`', () => {
        beforeEach(() => {
          two.trigger('click')
        })

        it('should add `Two` to value', () => {
          expect(onValueChange).toHaveBeenCalledWith(['One', 'Two'])
        })

        describe('and clicking `Two` again`', () => {
          beforeEach(() => {
            two.trigger('click')
          })

          it('should change value to `One`', () => {
            expect(onValueChange).toHaveBeenCalledWith(['One'])
          })
        })
      })
    })
  })
})

describe('okuToggleGroup Stories', () => {
  describe('single', () => {
    let wrapper: VueWrapper<InstanceType<typeof Single>>

    beforeEach(async () => {
      wrapper = mount(Single, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('vertical', () => {
    let wrapper: VueWrapper<InstanceType<typeof Vertical>>

    beforeEach(async () => {
      wrapper = mount(Vertical, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('multiple', () => {
    let wrapper: VueWrapper<InstanceType<typeof Multiple>>

    beforeEach(async () => {
      wrapper = mount(Multiple, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('chromatic', () => {
    let wrapper: VueWrapper<InstanceType<typeof Chromatic>>

    beforeEach(async () => {
      wrapper = shallowMount(Chromatic, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
