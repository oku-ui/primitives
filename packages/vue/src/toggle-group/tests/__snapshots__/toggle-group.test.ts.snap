// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`okuToggleGroup > should render OkuToggleGroup correctly 1`] = `"<oku-toggle-group-stub></oku-toggle-group-stub>"`;

exports[`okuToggleGroup > should render OkuToggleGroup correctly 2`] = `""`;

exports[`okuToggleGroup > should render OkuToggleGroupItem correctly 1`] = `[Error: \`OkuToggleGroupItem\` must be used within \`OkuToggleGroup\`]`;

exports[`okuToggleGroup Stories > chromatic > should render correctly 1`] = `
"<h1>Single</h1>
<h2>Off</h2>
<oku-toggle-group-stub type="single" class="toggle-group"></oku-toggle-group-stub>
<h2>On</h2>
<oku-toggle-group-stub type="single" class="toggle-group" default-value="1"></oku-toggle-group-stub>
<h2>Disabled</h2>
<oku-toggle-group-stub type="single" class="toggle-group" disabled=""></oku-toggle-group-stub>
<h1>Multiple</h1>
<h2>Off</h2>
<oku-toggle-group-stub type="multiple" class="toggle-group"></oku-toggle-group-stub>
<h2>One on</h2>
<oku-toggle-group-stub type="multiple" class="toggle-group" default-value="1"></oku-toggle-group-stub>
<h2>One and two on</h2>
<oku-toggle-group-stub type="multiple" class="toggle-group" default-value="1,2"></oku-toggle-group-stub>
<h2>Disabled</h2>
<oku-toggle-group-stub type="multiple" class="toggle-group" disabled=""></oku-toggle-group-stub>
<h1>Direction</h1>
<h2>Prop</h2>
<oku-toggle-group-stub type="single" class="toggle-group" default-value="1" dir="rtl"></oku-toggle-group-stub>
<h2>Inherited</h2>
<direction-provider-stub dir="rtl"></direction-provider-stub>
<h1>State attributes</h1>
<h2>Group disabled</h2>
<oku-toggle-group-stub type="multiple" class="toggle-group" default-value="1,2" disabled=""></oku-toggle-group-stub>
<h2>Group enabled with button override</h2>
<oku-toggle-group-stub type="multiple" class="toggle-group" default-value="['1', '2']" disabled="false"></oku-toggle-group-stub>
<h2>Group disabled with button override</h2>
<oku-toggle-group-stub type="multiple" class="toggle-group" default-value="['1', '2']" disabled="true"></oku-toggle-group-stub>"
`;

exports[`okuToggleGroup Stories > multiple > should render correctly 1`] = `
"<h1>Uncontrolled</h1>
<div class="toggle-group" aria-label="Options" role="group" dir="ltr" tabindex="0" style="outline: none;"><button type="button" aria-pressed="true" data-state="on" class="toggle-group-item" tabindex="-1" data-oku-collection-item=""> Option 1 </button><button type="button" aria-pressed="false" data-state="off" class="toggle-group-item" tabindex="-1" data-oku-collection-item=""> Option 2 </button><button type="button" aria-pressed="false" data-state="off" class="toggle-group-item" tabindex="-1" data-oku-collection-item=""> Option 3 </button></div>
<h1>Controlled</h1>
<div class="toggle-group" aria-label="Options" role="group" dir="ltr" tabindex="0" style="outline: none;"><button type="button" aria-pressed="false" data-state="off" class="toggle-group-item" tabindex="-1" data-oku-collection-item=""> Option 1 </button><button type="button" aria-pressed="false" data-state="off" class="toggle-group-item" tabindex="-1" data-oku-collection-item=""> Option 2 </button><button type="button" aria-pressed="false" data-state="off" class="toggle-group-item" tabindex="-1" data-oku-collection-item=""> Option 3 </button></div>"
`;

exports[`okuToggleGroup Stories > single > should render correctly 1`] = `
"<h1>Uncontrolled</h1>
<div class="toggle-group" aria-label="Options" role="group" dir="ltr" tabindex="0" style="outline: none;"><button type="button" data-state="on" role="radio" aria-checked="true" class="toggle-group-item" tabindex="-1" data-oku-collection-item=""> Option 1 </button><button type="button" data-state="off" role="radio" aria-checked="false" class="toggle-group-item" tabindex="-1" data-oku-collection-item=""> Option 2 </button><button type="button" data-state="off" role="radio" aria-checked="false" class="toggle-group-item" tabindex="-1" data-oku-collection-item=""> Option 3 </button></div>
<h1>Controlled</h1>
<div class="toggle-group" aria-label="Options" role="group" dir="ltr" tabindex="0" style="outline: none;"><button type="button" data-state="off" role="radio" aria-checked="false" class="toggle-group-item" tabindex="-1" data-oku-collection-item=""> Option 1 </button><button type="button" data-state="off" role="radio" aria-checked="false" class="toggle-group-item" tabindex="-1" data-oku-collection-item=""> Option 2 </button><button type="button" data-state="off" role="radio" aria-checked="false" class="toggle-group-item" tabindex="-1" data-oku-collection-item=""> Option 3 </button></div>"
`;

exports[`okuToggleGroup Stories > vertical > should render correctly 1`] = `"<div class="toggle-group" aria-label="Options" role="group" dir="ltr" tabindex="0" data-orientation="vertical" style="outline: none;"><button type="button" data-state="on" role="radio" aria-checked="true" class="toggle-group-item" tabindex="-1" data-orientation="vertical" data-oku-collection-item=""> Option 1 </button><button type="button" data-state="off" role="radio" aria-checked="false" class="toggle-group-item" tabindex="-1" data-orientation="vertical" data-oku-collection-item=""> Option 2 </button><button type="button" data-state="off" role="radio" aria-checked="false" class="toggle-group-item" tabindex="-1" data-orientation="vertical" data-oku-collection-item=""> Option 3 </button></div>"`;
